esp/bindings/http/platform/httpservice.cpp

```diff
@@ -184,6 +184,34 @@ void checkSetCORSAllowOrigin(EspHttpBinding *binding, CHttpRequest *req, CHttpRe
 
 int CEspHttpServer::processRequest()
 {
+    TraceManager traceManager("esp"); //we'd use an appropriate module/lib name here
+    auto tracer = traceManager.getTracer();
+
+    //Extract parent(caller) context from http header, likely done earlier in the process
+    //We'd need a setParentContextFromHeaders version supporting httptransport's StringArray  m_headers;
+    //TraceManager::setParentContextFromHeaders(const_cast<std::map<std::string, std::string> &>(request.headers), options);
+    //or labda function to extract parent context from http header
+
+    //Options used to annotate span representing the processing of http requests
+    opentelemetry::trace::StartSpanOptions options;
+    options.kind = opentelemetry::trace::SpanKind::kServer;
+
+    //Declare the span, provide appropriate attributes, and options
+    //Trace ID generated if no parent context is provided
+    auto processingRequestSpan =
```

Comment:
I would expect much of this code to be encapsulated inside jtrace.  That may not be possible, but having open telemetry code that manages spans scattered through the code suggests poor encapsulation.